### PR TITLE
Update test projects to netcoreapp3.0 and remove netcoreapp2.1 install from build script

### PR DESCRIPTION
**Pull request by ahsonkhan**

### CODE REVIEW

**Comment by ahsonkhan** [comment]
The tests should pass after the following fix: https://github.com/dotnet/coreclr/pull/19904

**Comment by ahsonkhan** [comment]
@dotnet-bot test this please

File: tools/dependencies.props [hunk 1]
```diff
@@ -10,6 +10,6 @@
     <LibuvVersion>1.9.1</LibuvVersion>
     <TestSdkVersion>15.0.0</TestSdkVersion>
     <XunitVersion>2.2.0</XunitVersion>
-    <BenchmarkDotNetVersion>0.11.1</BenchmarkDotNetVersion>
+    <BenchmarkDotNetVersion>0.11.1.776</BenchmarkDotNetVersion>
```
Comment (danmosemsft) [inline tools/dependencies.props]: Why this particular build

Comment (ahsonkhan) [reply tools/dependencies.props]: It's the latest BDN build available (which contains support for netcoreapp3.0).

**Comment by danmosemsft** [comment]
@ahsonkhan can this be merged now?

**Comment by ahsonkhan** [comment]
> can this be merged now?

I have one pending issue that I need to validate before we can merge this. I noticed some perf benchmark discrepancy on the latest run-time that I want to confirm first. Thanks for the ping.

File: src/System.Buffers.Experimental/System/Range.cs [hunk 2]
```diff
@@ -9,7 +9,7 @@
 namespace System.Buffers.Experimental
 {
     // TODO: consider allowing Last > First. Ennumeration will count down.
-    public readonly struct Range : IEnumerable<int>
+    public readonly struct RangeLab : IEnumerable<int>
```
Comment (ahsonkhan) [inline src/System.Buffers.Experimental/System/Range.cs]: As a quick (temporary) fix, I renamed the Range type within corefxlab to RangeLab. We need to remove it and update to use the new System.Range type: https://github.com/dotnet/corefxlab/issues/2628

cc @tarekgh

File: tests/System.Security.Cryptography.Asn1.Experimental.Tests/System.Security.Cryptography.Asn1.Experimental.Tests.csproj [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="..\..\tools\common.props" />
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <TargetFramework>netcoreapp3.0</TargetFramework>
```
Comment (ahsonkhan) [inline tests/System.Security.Cryptography.Asn1.Experimental.Tests/System.Security.Cryptography.Asn1.Experimental.Tests.csproj]: @bartonjs, is this fine? Also, the TFM here needs to be corrected to `netcoreapp2.1`: https://github.com/dotnet/corefxlab/blob/57be99a176421992e29009701a99a370983329a6/src/System.Security.Cryptography.Asn1.Experimental/System.Security.Cryptography.Asn1.Experimental.csproj#L5?

**Pull request merged**